lib/api/api.go

```diff
@@ -1472,6 +1476,10 @@ func (s *service) postSystemUpgrade(w http.ResponseWriter, _ *http.Request) {
 	opts := s.cfg.Options()
 	rel, err := upgrade.LatestRelease(opts.ReleasesURL, build.Version, opts.UpgradeToPreReleases)
 	if err != nil {
+		if err == upgrade.ErrUpgradeUnsupported {
+			http.Error(w, upgrade.ErrUpgradeUnsupported.Error(), http.StatusNotImplemented)
+			return
+		}
 		l.Warnln("getting latest release:", err)
 		http.Error(w, err.Error(), 500)
```

Comment:
Maybe break this whole thing into a `func httpError(w, err)` that does the right thing and can be reused below, plus take the opportunity to make the `500` literal `http.StatusInternalServerError`